src/lib.rs

```diff
@@ -9,24 +7,29 @@ extern crate hmac;
 extern crate nanoid;
 
 #[macro_export]
-/// To quickly create a hashmap.
-/// Example: `hashmap!["action"=>"query","meta"=>"siteinfo","siprop"=>"general|namespaces|namespacealiases|libraries|extensions|statistics"]`
-macro_rules! hashmap {
+/// To quickly create a param set.
+/// Example: `params!["action"=>"query","meta"=>"siteinfo","siprop"=>"general|namespaces|namespacealiases|libraries|extensions|statistics"]`
+macro_rules! params {
     ($( $key: expr => $val: expr ),* $(,)?) => {{
-         let mut map = ::std::collections::HashMap::new();
+         let mut map = $crate::Params::new();
          $( map.insert($key.into(), $val.into()); )*
          map
     }}
 }
 
-pub type Params = HashMap<String, String>;
-
 pub use reqwest;
 
 pub mod api;
 pub mod api_sync;
 pub mod error;
 pub mod method;
 pub mod page;
+pub mod params;
 pub mod title;
 pub mod user;
+
+pub use method::Method;
+pub use page::Page;
+pub use params::Params;
+pub use title::Title;
+pub use user::User;
```

Comment:
I'm not sure I'm a fan of re-exporting stuff here. The modules in question certainly have a tiny surface each, so I can see the appeal of doing it, but I'm not entirely convinced.

Reply:
I'll remove it then.